### ISSUE TITLE
SQL queries not being logged on AI for Azure Cloud Services

### ISSUE THREAD

**Issue by joaopgrassi** [microsoft/ApplicationInsights-dotnet#931]
## Repro Steps
I enabled Application Insights on an existing Cloud Services web and worker role via adding the SDK and setting the Instrumentation Key. I can see the dependencies (SQL) being logged, but I don't see actual SQL queries.  I have read a lot of resources, including this issue #768, but everywhere it mentions only Web Apps, Vms or App Services. There's no much documentation for Azure Cloud Services and the ones I found are somewhat confusing:

1. [This one](https://github.com/uglide/azure-content/blob/master/articles/application-insights/app-insights-cloudservices.md#dependencies) says if my app uses .NET framework 4.6 I don't have to do anything, but below it says that I need to set up the Status Monitor. Which one is true here? Do I **really** need to add those scripts into my app?

2. This link: [Set up dependency monitoring](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-asp-net-dependencies#set-up-dependency-monitoring) says that I either need to install .NET Framework 4.6+ (which is there because I ran on osFamily=5) or "Set up startup task" which the link there redirects me to nothing.

3. I executed the query mentioned on #768, and I can see it's not configured correctly, because the I get **rddf**  for the sdkVersion.

##  Actual Behavior
Everything is logged correctly on Application Insights (Requests, Traces, Dependencies), but I don't see SQL Queries on SQL Server dependencies.

## Expected Behavior
SQL Queries are displayed on SQL Server dependencies

## Version Info
**SDK Version** : rddf:2.7.2-19477
**.NET Version** :	4.6.2 						
**How Application was onboarded with SDK** : I installed the NuGet Packages and configured the InstrumentantionKey in the app manually (without wizard on VS) as instructed here: [Application Insights for Azure Cloud Services](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-cloudservices#sdk)
**Hosting Info**: Azure Cloud Services (osFamily=5). Web role and Worker Role
**Packages installed**:
```
<package id="Microsoft.ApplicationInsights" version="2.7.2" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.Agent.Intercept" version="2.4.0" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.DependencyCollector" version="2.7.2" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.PerfCounterCollector" version="2.7.2" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.Web" version="2.7.2" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.WindowsServer" version="2.7.2" targetFramework="net462" />
<package id="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" version="2.7.2" targetFramework="net462" />
```

**Comment by joaopgrassi** [comment]
Okay - got it working.

I've found this [Repo](https://github.com/uglide/azure-content/blob/master/articles/application-insights/app-insights-cloudservices.md#dependencies) after posting this here and followed the guide on installing the **Application Insights Agent** into my cloud services roles. 
Just looked now and I can see the queries now showing :). Also the sdk now reports as `rddp:2.7.2-19477`.

My question is now: Is there any impact on my apps with the Application Insights Agent enabled like that? Should I be aware of something when deploying this to production? 
Thanks!

**Comment by joaopgrassi** [comment]
@Dmitry-Matveev thanks for explaining. I've been running it since I opened the issue.. no issues so far :) For my original issue, we could close this, but there's a "deeper" question against the docs lacking this. I tried contacting people from the @docs via Twitter, but no one seemed to have followed up. You can check this thread in case you want: https://twitter.com/jotagrassi/status/1052667656908754944

**Comment by Dmitry-Matveev** [comment]
@joaopgrassi , it's very well hidden in the docs, but I was able to find it :)
Here is the breadcrumb:
- [Setup Dependency Monitoring Doc](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-asp-net-dependencies#set-up-dependency-monitoring)
- [Use startup task](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-cloudservices) links leads you to the full doc
- Where section [Sample app..](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-cloudservices#sample-application-instrumented-with-application-insights) and [More telemetry](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-cloudservices#more-telemetry) will refer the GitHub example with the instructions;
- For instance, [Reporting Dependencies](https://github.com/Microsoft/ApplicationInsights-Home/tree/master/Samples/AzureEmailService/MvcWebRole#reporting-dependencies) steps that mentions Agent and startup tasks..

Long story short - somewhat in docs, not fully, may not be able to find it unless you know it exists..

You can open issue right on Azure Docs GitHub, that team typically reacts to those quickly (you can do that using "Feedback" at the top of the doc pages), not sure about Twitter queue on that team currently...

**Comment by joaopgrassi** [comment]
@Dmitry-Matveev yeah, in the git repos `WorkerRoleA` and `WorkerRoleB` there are the steps on how to make it work. But I agree.. it's very deep into that, and I have read that "main" page at least 3 times and didn't "find" it :) 

Will create the issue later this week. Thank you for digging into this!

**Comment by Dmitry-Matveev** [comment]
@joaopgrassi, the GitHub readme's appear to be outdated a bit. Long time ago, the only sure method to collect dependencies was indeed to drop profiler DLL (either via Status Monitor or Web Site Extension) because .NET 4.5 didn't have full instrumentation with Event Source and didn't emit all the necessary messages for AI to track all SQL and Http dependencies reliably.

.NET 4.6.X has those places covered with ES messages, so profiler is not necessary unless you'd like to get SQL query. Further along in .NET and .NET Core, Diagnostic Source messages are emitted by framework for SQL/Http calls and AI subscribes to those to report dependencies.

**Comment by ackava** [comment]
Problem is with Microsoft Monitoring Agent, it installs profiler and AI cannot install new profiler to monitor sql queries.

Removing this agent makes sql queries appear but after server is restarted or vm is reprovisioned, Azure installs Microsoft Monitoring Agent again.

This issue with workaround was reported on this thread but no work was done, https://github.com/microsoft/ApplicationInsights-dotnet-server/issues/922

**Comment by tcihak-fqa** [comment]
Unfortunately we are still using the legacy cloud services.
How does one remove the Microsoft Monitoring Agent?
Because I followed these instructions and that didn't work.
https://docs.microsoft.com/en-us/azure/azure-monitor/app/cloudservices#set-up-status-monitor-to-collect-full-sql-queries-optional

We've hit so many minefields trying to get AI collection configured correctly in .NET and javascript.
It's so frustrating.

**Comment by TimothyMothra** [comment]
Our team doesn't own the Microsoft Monitoring Agent.
@cijothomas, @ank3it do you know who would be able to help here?
